stdlib/public/core/FloatingPoint.swift.gyb

```diff
@@ -2461,7 +2461,13 @@ where Self.RawSignificand : FixedWidthInteger {
 %     end
     }
     let unitRandom = Self.init(rand) * Self.ulpOfOne / 2
-    return delta * unitRandom + range.lowerBound
+    let randFloat = delta * unitRandom + range.lowerBound
+%   if 'Closed' not in Range:
+    if randFloat == range.upperBound {
+      return randFloat.nextDown
+    }
+%   end
```

Comment:
This introduces a (very slight) bias, as it effectively makes the penultimate value 1.5 times more likely to occur than every other value in the binade. As a hacky quick fix to preserve the desired invariant, it's semi-acceptable, but let's see if we can get a better long-term fix done instead.

Reply:
I'll either make a patch myself or send you some notes on how to do this in a more principled way later today.

Reply:
Sounds good to me. Would love to get some notes even if you patch it yourself if that's fine.

Reply:
If we re-draw with the same max rng, wouldn't we be forever re-drawing since upperBound would always be occurring?

Reply:
Yeah, that is a concern, though it won't happen with non-pathological RNGs =)

Maybe for the test use an RNG that instead generates `[.min, .max, basic LCG sequence]`.